### DISCUSSION TITLE
Is there a text version tool can convert text to scratch blocks?

### DISCUSSION THREAD

**Discussion by OriLiMu** [ScratchAddons/ScratchAddons#7660] [Discussion]
When you want to make something complicated, better idea to make it in the text version. So I wonder there is something can make it? 
I found 
https://tosh.blob.codes/app/ which is for scratch 2.0 
and 
https://github.com/brianyu28/scratchtext. Seems abandoned. 

# Update 
- The meaning of this I think is just tired of the **drag block** work which is better for kid but terrible work for adults who want to make some complicated things. 
- I see that the sb3 file can be unzipped. There is a json file contains all the code.  Find a way to edit it nicely would be a way to make it. 
- The tosh is almost what I want. but in scratch 2.0
- I though that may be no need to create a prefect version of this. A workaround which I think is workable and much less work to do is edit the code outside scratch editor then translate the code to the json mode the zip it

**Comment by Samq64** [comment]
I don't think so but I haven't really looked into it. There's https://leopardjs.com/ but it probably can't be used to convert back to a Scratch project.

Related:
- #6922
- #5157

**Comment by StickmanRed** [comment]
How about boxes of text-based code within the Scratch block workspace? The code could be stored in comments... I don't think it would be a problem for non-users, since it could possibly function as a code concept.
To run the code, there would be an option to compile and insert the code into the actual stacks; to edit the code as text, blocks would be selected and spliced from the stack, and a comment would be made containing the typed representation.

**Reply by OriLiMu** [reply]
I am not sure if I follow your saying.
what do you mean the "boxes of text-based code"? 
I don't think it is a good way to save the code in the comment hard to read and edit. 
I think the **tosh** which I mentioned in the first post is the best way to make it.

**Reply by StickmanRed** [reply]
I'm not sure if something similar to tosh could be a Scratch addon, more like a separate development. However, I agree that it would be a useful tool.

About my comment, I was thinking of a text area (maybe resizable or draggable) in which you can type Scratch code. I think this type would fit with Scratch's multiple-stack style

**Comment by bit-turtle** [comment]
I am working on a project called "lascra" which is a text representation of scratch code that can be compiled into sprite3 files. [bit-turtle/lascra](https://github.com/bit-turtle/lascra)
It works by modifying the sprite.json file inside of the exported sprite3 files.
Currently I don't have any releases published for it, but you can build it with CMake.

**Comment by bit-turtle** [comment]
@OriLiMu

My text to scratch compiler is in a good state now, what do you think?

You can try it out here: [bit-turtle.github.io/lascra.html](https://bit-turtle.github.io/lascra.html)

**Comment by Zhiyuanbu** [comment]
View `https://scratch.mit.edu/discuss/topic/871013/?page=1#post-9027171`
